compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -976,20 +976,42 @@ object Trees {
   def genericEmptyTree[T <: Untyped]: Thicket[T]        = theEmptyTree.asInstanceOf[Thicket[T]]
 
   /** Tree that replaces a level 1 splices in pickled (level 0) quotes.
-   *  It is only used when picking quotes (will never be in a TASTy file).
+   *  It is only used when encoding pickled quotes. These will be encoded
+   *  as TastyQuoteHole when pickled.
```

Comment:
I have read the documentation of Hole and TastyQuoteHole twice but I'm still confused on which one is used when, could this be described in more details, or could the documentation link to some other part of the codebase where this is explained?

Reply:
Updated the documentation